### PR TITLE
OSSMDOC-463 Verify sidecar injection.

### PR DESCRIPTION
**Pull request by JStickler**
This PR 
* adds a new topic (Validating sidecar injection)
* changes the example to use the bookinfo sample application.
* updates some steps for missing content or clarity.
* updates a broken cross-reference and adds a cross-reference to the viewing logs topic.

Preview -> https://deploy-preview-43238--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/prepare-to-deploy-applications-ossm.html
AND   
https://deploy-preview-43238--osdocs.netlify.app/openshift-enterprise/latest/service_mesh/v2x/ossm-vs-community.html#ossm-automatic-injection_ossm-vs-istio

Eng review - lucasponce, jmazzitelli
QE review - skondkar
Peer review - nalhadef

### CODE REVIEW

File: modules/ossm-automatic-sidecar-injection.adoc [hunk 1]
```diff
@@ -7,63 +7,76 @@
 [id="ossm-automatic-sidecar-injection_{context}"]
 = Enabling automatic sidecar injection
 
-When deploying an application, you must opt-in to injection by setting the `sidecar.istio.io/inject` annotation to `"true"`. Opting in ensures that the sidecar injection does not interfere with other {product-title} features such as builder pods used by numerous frameworks within the {product-title} ecosystem.
+When deploying an application, you must opt-in to injection by setting the `sidecar.istio.io/inject` annotation to `true` in the `deployment` object. Opting in ensures that the sidecar injection does not interfere with other {product-title} features such as builder pods used by numerous frameworks within the {product-title} ecosystem.
```
Comment (lucasponce) [inline modules/ossm-automatic-sidecar-injection.adoc:10]: The annotation should go to the pod, so, it should be in the spec.template section not in the "deployment" section:
image

File: modules/ossm-sidecar-validate-kiali.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+* Versioned app graph
+* Workload graph
+
+The *Applications* page displays a *Missing Sidecar* icon in the *Details* column for any applications in a namespace that do not have a sidecar.
+
+The *Workloads* page displays a *Missing Sidecar* icon in the *Details* column for any applications in a namespace that do not have a sidecar.
+
+The *Services* page displays a *Missing Sidecar* icon in the *Details* column for any applications in a namespace that do not have a sidecar. Note that when there are multiple versions of a service, you use the *Service Details* page to view *Missing Sidecar* icons.
+
+The *Workload Details* page has a special unified *Logs* tab that lets you view and correlate application and proxy logs. You can view the Envoy logs as another way to validate sidecar injection for your application workloads.
```
Comment (lucasponce) [inline modules/ossm-sidecar-validate-kiali.adoc:23]: Worth mentioning that the "Workload Details" offer an "Envoy" tab where user can navigate in the details of the proxy:
https://kiali.io/docs/features/details/#envoy

File: modules/ossm-sidecar-validate-kiali.adoc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+The Kiali console offers several ways to validate whether or not your applications, services, and workloads have a sidecar proxy.
+
+The *Graph* page displays a node badge indicating a *Missing Sidecar* on the following graphs:
+
+* App graph
+* Versioned app graph
+* Workload graph
+
+The *Applications* page displays a *Missing Sidecar* icon in the *Details* column for any applications in a namespace that do not have a sidecar.
```
Comment (lucasponce) [inline modules/ossm-sidecar-validate-kiali.adoc:17]: I miss screenshots. It's hard to follow these without a picture.

The Kiali project is a dashboard / User Interface and not having any pictures is something odd to me.

Comment (JStickler) [reply modules/ossm-sidecar-validate-kiali.adoc:17]: Yeah, I think the docs team needs to have a discussion around how to add at least a few screenshots.  The problem is, they have to be localized, so they need to be reproducable for the translation team.

File: modules/ossm-vs-istio.adoc [hunk 4]
```diff
@@ -29,6 +29,27 @@ The upstream Istio community installation automatically injects the sidecar into
...
+
+|Pod Label
+|supports "true" and "false"
+|not supported
+
+|Pod Annotation
+|supports "false" only
+|"true" and "false"
+|===
+
```
Comment (jmazzitelli) [inline modules/ossm-vs-istio.adoc:52]: This is how I saw OSSM 2.1 work last November - specifically, that upstream Istio supports auto-injection on a per-namespace basis (using the namespace label) whereas OSSM only supports allowing auto-injection on pods only (using the pod annotation only).

It has come to my attention that this may no longer be the case in OSSM. We should confirm that this is how it is supposed to be. It could be that OSSM did not intend to allow for namespace auto-injection label to take effect but a bug has enabled it. This is just a guess. But I was under the impression OSSM was not allowing namespace auto-injection labels to be supported for security purposes.

So, we should at least get confirmation as to what is the expected behavior of OSSM 2.2. Is it:

(a) OSSM supports auto-injection on a per-namespace basis (via namespace label)

or

(b) OSSM does NOT support auto-injection per-namespace but ONLY supports auto-injection on a per-pod basis (via the pod annotation, and perhaps via a pod label, too?)

File: modules/ossm-sidecar-validate-kiali.adoc (hunk 3 above)
Comment (lucasponce) [reply modules/ossm-sidecar-validate-kiali.adoc:17]: I can attach other examples from Openshift documentation where there are screenshots with draconian rules

File: modules/ossm-automatic-sidecar-injection.adoc (hunk 1 above)
Comment (JStickler) [reply modules/ossm-automatic-sidecar-injection.adoc:10]: Lucas, I can't tell from the screen shot, is this a deployment.yaml file or a pod.yaml file?  It looks like it matches an object of `kind:Deployment` ,which is what I've got documented in this topic.

Comment (lucasponce) [reply modules/ossm-automatic-sidecar-injection.adoc:10]: It's a deployment. The template section will tell which annotations will go directly on the generated pods.

Comment (JStickler) [reply modules/ossm-automatic-sidecar-injection.adoc:10]: Updating this to include the full path (`spec.template.metadata.annotations.sidecar.istio.io/inject`) for clarity.

File: modules/ossm-automatic-sidecar-injection.adoc [hunk 5]
```diff
@@ -7,63 +7,76 @@
 [id="ossm-automatic-sidecar-injection_{context}"]
 = Enabling automatic sidecar injection
 
-When deploying an application, you must opt-in to injection by setting the `sidecar.istio.io/inject` annotation to `"true"`. Opting in ensures that the sidecar injection does not interfere with other {product-title} features such as builder pods used by numerous frameworks within the {product-title} ecosystem.
+When deploying an application, you must opt-in to injection by configuring the setting `spec.template.metadata.annotations.sidecar.istio.io/inject` to `true` in the `deployment` object. Opting in ensures that the sidecar injection does not interfere with other {product-title} features such as builder pods used by numerous frameworks within the {product-title} ecosystem.
```
Comment (jmazzitelli) [inline modules/ossm-automatic-sidecar-injection.adoc:10]: This kinda makes it as confusing because "." is in the name of the annotation so `spec.template.metadata.annotations.sidecar.istio.io/inject` is verbose (what's the name of the annotation? You have to stare at that for a while to see it). Perhaps this for better clarity:

```suggestion
When deploying an application, you must opt-in to injection by configuring the annotation `sidecar.istio.io/inject` in `spec.template.metadata.annotations` to `true` in the `deployment` object. Opting in ensures that the sidecar injection does not interfere with other {product-title} features such as builder pods used by numerous frameworks within the {product-title} ecosystem.
```

File: modules/ossm-vs-istio.adoc (hunk 4 above)
Comment (JStickler) [reply modules/ossm-vs-istio.adoc]: I'm commenting out this table so that we can move forward while waiting on an answer to the question of what we support.

File: modules/ossm-automatic-sidecar-injection.adoc [hunk 6]
```diff
@@ -7,63 +7,76 @@
...
++
+[source,terminal]
+----
+oc get deployment -n bookinfo ratings-v1 -o yaml
+----
++
+. Open the application's deployment configuration YAML file in an editor.
 
-. Add `sidecar.istio.io/inject` to the configuration YAML with a value of `"true"` in the `spec.template.metadata.annotations.sidecar.istio/inject` field. See the following example for an app called `sleep`.
+. Add `spec.template.metadata.annotations.sidecar.istio/inject` to your Deployment YAML and set `sidecar.istio.io/inject` to `true` as shown in the following example.
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-automatic-sidecar-injection.adoc:34]: I took the first suggestion, but this one ends up being really awkward. I'm leaving it as is and hoping that the YAML example right below makes it clear what's going on.

**Comment by skondkar** [comment]
Verified with the steps in preview doc. LGTM

File: modules/ossm-sidecar-validate-kiali.adoc [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+* service_mesh/v2x/prepare-to-deploy-applications-ossm.adoc
+////
+:_content-type: CONCEPT
+[id="ossm-validating-sidecar_{context}"]
+= Validating sidecar injection
+
+The Kiali console offers several ways to validate whether or not your applications, services, and workloads have a sidecar proxy.
+
+.Missing sidecar badge
+image::ossm-node-badge-missing-sidecar.svg[Missing Sidecar badge]
```
Comment (nalhadef) [inline modules/ossm-sidecar-validate-kiali.adoc:12]: Just to be sure, the image in the preview is the missing sidecar badge and not the system telling me the image is missing. 
image

File: modules/ossm-sidecar-validate-kiali.adoc [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+The *Graph* page displays a node badge indicating a *Missing Sidecar* on the following graphs:
+
+* App graph
+* Versioned app graph
+* Workload graph
+
+.Missing sidecar icon
+image::ossm-icon-missing-sidecar.png[Missing Sidecar icon]
```
Comment (nalhadef) [inline modules/ossm-sidecar-validate-kiali.adoc:21]: Same as above.

image

File: modules/ossm-sidecar-validate-kiali.adoc [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+The *Applications* page displays a *Missing Sidecar* icon in the *Details* column for any applications in a namespace that do not have a sidecar.
+
+The *Workloads* page displays a *Missing Sidecar* icon in the *Details* column for any applications in a namespace that do not have a sidecar.
+
+The *Services* page displays a *Missing Sidecar* icon in the *Details* column for any applications in a namespace that do not have a sidecar. Note that when there are multiple versions of a service, you use the *Service Details* page to view *Missing Sidecar* icons.
```
Comment (nalhadef) [inline modules/ossm-sidecar-validate-kiali.adoc:27]: For the second sentence: Use admonition formatting or reword to avoid "Note that."

File: modules/ossm-sidecar-validate-kiali.adoc [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+The *Workload Details* page has a special unified *Logs* tab that lets you view and correlate application and proxy logs. You can view the Envoy logs as another way to validate sidecar injection for your application workloads.
+
+The *Workload Details* page also has an *Envoy* tab for any workload that is an Envoy proxy or has been injected with an Envoy proxy. This tab displays a built-in Envoy dashboard that includes subtabs for *Clusters*, *Listeners*, *Routes*, *Bootstrap*, *Config*, and *Metrics*.
```
Comment (nalhadef) [inline modules/ossm-sidecar-validate-kiali.adoc:31]: Suggestion: Make the subtabs a list.

File: modules/ossm-vs-istio.adoc [hunk 11]
```diff
@@ -29,6 +29,29 @@ The upstream Istio community installation automatically injects the sidecar into
 
 {SMProductName} does not automatically inject the sidecar to any pods, but requires you to opt in to injection using an annotation without labeling projects. This method requires fewer privileges and does not conflict with other OpenShift capabilities such as builder pods. To enable automatic injection you specify the `sidecar.istio.io/inject` annotation as described in the Automatic sidecar injection section.
 
+////
```
Comment (nalhadef) [inline modules/ossm-vs-istio.adoc:32]: This relates to Line 5: Need to add the content type attribute.

**Review by nalhadef — COMMENTED** [review]
Don't forget to squash. I see 5 commits currently.

File: modules/ossm-automatic-sidecar-injection.adoc [hunk 12]
```diff
@@ -7,63 +7,76 @@
...
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-automatic-sidecar-injection.adoc:34]: In this case, since I'm talking about editing an actual Deployment file, I believe that the capitalization is correct (it's lowercase for general references).

See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

[the remark this answers is not in the document]
Comment (nalhadef) [reply modules/ossm-automatic-sidecar-injection.adoc:34]: But......Is the name of the file Deployment? If not, it's a general use of the word. Like: "Create a deployment file named `ShipIt.abc`.

File: modules/ossm-kiali-viewing-logs.adoc [hunk 13]
```diff
@@ -31,6 +30,11 @@ The Kiali Overview page displays namespaces that have been added to the mesh tha
 
 . On the *Workloads* page, select the project from the *Namespace* menu.
 
-. If necessary, use the filter to find the workload whose logs you want to view.  Click the workload *Name*.
+. If necessary, use the filter to find the workload whose logs you want to view.  Click the workload *Name*.  For example, click *ratings-v1*.
```
[the remark this answers is not in the document]
Comment (JStickler) [reply modules/ossm-kiali-viewing-logs.adoc:33]: In this case *Name* is a column heading, which is a GUI item, and should be formatted as bold according to 
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference

File: modules/ossm-sidecar-validate-kiali.adoc (hunk 7 above)
Comment (JStickler) [reply modules/ossm-sidecar-validate-kiali.adoc:12]: Yes, the purple badge is the icon.  =)

File: modules/ossm-automatic-sidecar-injection.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (nalhadef) [reply modules/ossm-automatic-sidecar-injection.adoc:34]: OK, if this is an API object, I guess it is capitalized. :-)

File: modules/ossm-sidecar-validate-kiali.adoc (hunk 8 above)
Comment (JStickler) [reply modules/ossm-sidecar-validate-kiali.adoc:21]: Yes, I'm laughing because this could be confusing to review if you don't know the GUI!

File: modules/ossm-kiali-viewing-logs.adoc (hunk 13 above)
[the remark this answers is not in the document]
Comment (nalhadef) [reply modules/ossm-kiali-viewing-logs.adoc:33]: Is workload the same as column heading? If so, I agree with you. If not, you can rewrite it to refer explicitly to the column heading.
